src/imitation/data/types.py

```diff
@@ -69,26 +69,53 @@ def __post_init__(self):
 
 
 @dataclasses.dataclass(frozen=True)
-class Transitions:
-    """A batch of obs-act-obs-done transitions.
+class TransitionsMinimal:
+    """A batch of obs-act transitions.
 
-    Often generated by combining and processing several TrajectoryNoRew objects via
-    `flatten_trajectories()`.
+    This class and its subclasses are usually instantiated via
+    `imitation.data.rollout.flatten_trajectories`.
     """
 
     obs: np.ndarray
     """
     Previous observations. Shape: (batch_size, ) + observation_shape.
 
     The i'th observation `obs[i]` in this array is the observation seen
-    by the agent when choosing action `acts[i]`.
-
-    Invariant: obs.dtype == next_obs.dtype.
```

Comment:
Shouldn't this invariant be documented somewhere (in `Transitions` now, I guess?)

Reply:
Accidental deletion. Thanks

Reply:
`TransitionsMinimal` doesn't have `next_obs`, so I think this belongs in `Transitions` now instead